modules/publicize/publicize-jetpack.php

```diff
@@ -390,6 +390,225 @@ function get_services( $filter = 'all' ) {
 		}
 	}
 
+	/**
+	 * Retrieves current list of connections and applies filters.
+	 *
+	 * Retrieves current available connections and checks if the connections
+	 * have already been used to share current post. Finally, the checkbox
+	 * form UI fields are calculated. This function exposes connection form
+	 * data directly as array so it can be retrieved for static HTML generation
+	 * or JSON consumption.
+	 *
+	 * @since 5.9.1
+	 *
+	 * @param integer $post_id Optional. Post ID to query connection status for: will use current post if missing.
+	 *
+	 * @return array {
+	 *     Array of UI setup data for connection list form.
+	 *
+	 *     @type string 'unique_id'       ID string representing connection
+	 *     @type bool   'checked'         Default value of checkbox for connection.
+	 *     @type bool   'disabled'        String of HTML disabled property of checkbox. Empty if not disabled.
+	 *     @type bool   'active'          True if connection is not skipped by filters and is not already done.
+	 *     @type bool   'hidden_checkbox' True if the connection should not be shared to by current user.
+	 *     @type string 'label'           Text description of checkbox.
+	 *     @type string 'display_name'    Username for sharing account.
+	 * }
+	 */
+	public function get_filtered_connection_data( $post_id = null ) {
+		$connection_list = array();
+
+		$post = get_post( $post_id ); // Defaults to current post if $post_id is null.
```

Comment:
Do we need a `null` check in case there is no current post?

Reply:
Hmm. I need to work out what needs to happen in this case. 

I'm just thinking out loud here: Post ID is used to disable the connection checkbox if the post has already been shared to some or all connections. A null post ID may come in to play with a new post. I think that should just leave all connection checkboxes enabled...

I'm going to do some testing around this.

Reply:
This was indeed an issue. Just pushed in a fix with associated test cases. Thanks!